drivers/clock_control/clock_control_silabs_siwx91x.c

```diff
@@ -202,6 +220,24 @@ static int siwx91x_clock_init(const struct device *dev)
 	siwx91x_clock_on(dev, (clock_control_subsys_t)SIWX91X_CLK_I2C1);
 #endif
 
+#ifdef CONFIG_PM
+	/* Initialize platform-specific configurations. */
+	sli_si91x_platform_init();
+	/* Initialize the power manager */
+	sl_si91x_power_manager_init();
+
+	/* Remove peripheral requirements */
+	sl_si91x_power_manager_remove_peripheral_requirement(&peripheral_config);
+
+	/* Configure RAM retention settings */
+	sl_si91x_power_manager_configure_ram_retention(&ram_configuration);
+
+	/* Add a power state requirement for PS4 */
+	sl_si91x_power_manager_add_ps_requirement(SL_SI91X_POWER_MANAGER_PS4);
+
+	/* Set the clock scaling to performance mode */
+	sl_si91x_power_manager_set_clock_scaling(SL_SI91X_POWER_MANAGER_PERFORMANCE);
```

Comment:
The comments above each lines just reword the C code. You can probably remove them.